ksmt-symfpu/src/main/kotlin/org/ksmt/symfpu/FpToBvTransformer.kt

```diff
@@ -0,0 +1,236 @@
+package org.ksmt.symfpu
+
+import org.ksmt.KContext
+import org.ksmt.expr.KApp
+import org.ksmt.expr.KBvToFpExpr
+import org.ksmt.expr.KConst
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.KFpAbsExpr
+import org.ksmt.expr.KFpAddExpr
+import org.ksmt.expr.KFpDivExpr
+import org.ksmt.expr.KFpEqualExpr
+import org.ksmt.expr.KFpFromBvExpr
+import org.ksmt.expr.KFpFusedMulAddExpr
+import org.ksmt.expr.KFpGreaterExpr
+import org.ksmt.expr.KFpGreaterOrEqualExpr
+import org.ksmt.expr.KFpIsInfiniteExpr
+import org.ksmt.expr.KFpIsNaNExpr
+import org.ksmt.expr.KFpIsNegativeExpr
+import org.ksmt.expr.KFpIsNormalExpr
+import org.ksmt.expr.KFpIsPositiveExpr
+import org.ksmt.expr.KFpIsSubnormalExpr
+import org.ksmt.expr.KFpIsZeroExpr
+import org.ksmt.expr.KFpLessExpr
+import org.ksmt.expr.KFpLessOrEqualExpr
+import org.ksmt.expr.KFpMaxExpr
+import org.ksmt.expr.KFpMinExpr
+import org.ksmt.expr.KFpMulExpr
+import org.ksmt.expr.KFpNegationExpr
+import org.ksmt.expr.KFpRemExpr
+import org.ksmt.expr.KFpRoundToIntegralExpr
+import org.ksmt.expr.KFpSqrtExpr
+import org.ksmt.expr.KFpSubExpr
+import org.ksmt.expr.KFpToBvExpr
+import org.ksmt.expr.KFpToFpExpr
+import org.ksmt.expr.KFpToIEEEBvExpr
+import org.ksmt.expr.KFpValue
+import org.ksmt.expr.rewrite.simplify.simplifyFpToIEEEBvExpr
+import org.ksmt.expr.transformer.KNonRecursiveTransformer
+import org.ksmt.sort.KBoolSort
+import org.ksmt.sort.KBvSort
+import org.ksmt.sort.KFpSort
+import org.ksmt.sort.KSort
+import org.ksmt.utils.asExpr
+import org.ksmt.utils.cast
+
+class FpToBvTransformer(ctx: KContext) : KNonRecursiveTransformer(ctx) {
```

Comment:
Currently, there is a big issue with arrays and uninterpreted functions. Basically, when we have `(Array Fp Fp)` or `(Fp -> Fp -> Fp)` we must translate these declarations to `(Array Bv Bv)` and `(Bv -> Bv -> Bv)`.